src/Sarif/Baseline/V2/WhatComparer.cs

```diff
@@ -188,5 +189,29 @@ private static string GetFirstSnippet(ExtractedResult result)
 
             return null;
         }
+
+        private static string GetCanonicalizedMessage(ExtractedResult result)
+        {
+            string rawMessage = result.Result.GetMessageText(result.Result.GetRule(result.OriginalRun));
+
+            // Canonicalize the message by replacing any line numbers in it with consistent markers
+            Region firstRegion = result.Result?.Locations?.FirstOrDefault()?.PhysicalLocation?.Region;
+            if (firstRegion != null)
+            {
+                rawMessage = rawMessage
+                    .Replace(firstRegion.StartLine.ToString(), "~SL~")
```

Comment:
>Replace [](start = 21, length = 7)

Should we do a more precise match so if `StartLine` is 2 we don't replace "42" with "42~SL~"?

Reply:
I think it's better not to, actually. We'd likely need a Regex to describe the boundaries (non-digits around the number) which would be a lot slower, particularly if the message is huge.

If the Result doesn't move, even when the replacement is cutting up unrelated numbers, it'll do so the same way for the old and new Result, so they'll still match.

So the case where this hinders matching is when the Result moves and the line number before or after happened to be an unrelated number in the message.  If we see that happening too often, I can certainly revisit this.